components/script/dom/node.rs

```diff
@@ -782,8 +782,16 @@ impl Node {
             self.inclusive_descendants_version(),
             doc.inclusive_descendants_version(),
         ) + 1;
-        for ancestor in self.inclusive_ancestors(ShadowIncluding::No) {
-            ancestor.inclusive_descendants_version.set(version);
+
+        let mut parent = &self.parent_node;
+        while !parent.is_null() {
+            match parent.if_is_some(|p| {
```

Comment:
I think `while !parent.is_null()` could simply be replaced with a `loop`, since `parent.if_is_some` already checks for null.